maven-jxr-plugin/src/main/java/org/apache/maven/plugin/jxr/AbstractJxrReport.java

```diff
@@ -60,31 +57,9 @@
 public abstract class AbstractJxrReport
     extends AbstractMavenReport
 {
-    @Parameter( defaultValue = "${project}", readonly = true, required = true )
-    private MavenProject project;
 
-    @Component
-    private Renderer siteRenderer;
-
-    /**
-     * Output folder where the main page of the report will be generated. Note that this parameter is only relevant if
-     * the goal is run directly from the command line or from the default lifecycle. If the goal is run indirectly as
-     * part of a site generation, the output directory configured in the Maven Site Plugin will be used instead.
-     */
-    @Parameter( defaultValue = "${project.reporting.outputDirectory}", required = true )
-    private File outputDirectory;
-
-    /**
-     * File input encoding.
-     */
-    @Parameter( property = "encoding", defaultValue = "${project.build.sourceEncoding}" )
-    private String inputEncoding;
-
-    /**
-     * File output encoding.
-     */
-    @Parameter( property = "outputEncoding", defaultValue = "${project.reporting.outputEncoding}" )
-    private String outputEncoding;
+    @Parameter( defaultValue = "${session}", readonly = true, required = true )
+    private MavenSession session;
```

Comment:
where is used - field and getter method?

Reply:
Good question. This is used by UTs. Without it no repos can be injected. In next major Onwill pull it up.